generative_data_prep/data_prep/pipeline.py

```diff
@@ -269,8 +275,31 @@ def multiprocess_data_prep(
             )
         )
 
-    with Pool(num_workers) as p:
-        _ = p.starmap(data_prep_main, data_prep_main_args_list)
+    futures = []
+    with concurrent.futures.ProcessPoolExecutor(max_workers=num_workers) as executor:
+        futures = [executor.submit(data_prep_main_helper, args) for args in data_prep_main_args_list]
+
+    # if one process fails with an exception such as a RuntimeError, all other processes will fail with a
+    # BrokenProcessPool exception.  In such a situation we only want to show the user the RuntimeError.  We only want
+    # to show the user the BrokenProcessPool exception if all failing processes failed with this error.  This
+    # is why we have this complicated logic below.
+    broken_process_indices = []
+    broken_process_pool_exc = None
+    # search for any "interesting" exception (a non-BrokenProcessPool Exception)
+    for i, future in enumerate(futures):
+        try:
+            future.result()
+        except Exception as exc:
+            if isinstance(exc, concurrent.futures.process.BrokenProcessPool):
+                broken_process_indices.append(str(i))
+                broken_process_pool_exc = exc
+            else:
+                print(f'\n\nProcess {i} failed with the following exception:')
+                raise exc from None
+    # if no "interesting" exceptions are found, raise the BrokenProcessPool Exception
+    if len(broken_process_indices) > 0:
+        print(f'\n\nProcesses {", ".join(broken_process_indices)} failed with the following exception:')
+        raise broken_process_pool_exc
```

Comment:
Is this the case when a processes is Pkilled and it can't raise a proper exception? Or what case would this happen in?

Reply:
yup if you kill -9 one of the processes